app/src/main/java/org/jtaccuino/app/Studio.java

```diff
@@ -81,15 +81,38 @@ public Parent createParentNode() {
                 ToolBar.createMainToolBar()));
         bp.setBottom(StatusLine.getDefault().getNode());
 
-        sheetPane.getTabs().add(new TabSheet(SheetManager.getDefault().of(notebook)));
+        initTabs();
 
         return bp;
     }
 
-    public void shutdown() {
+    private void initTabs() {
+        var openFiles = FilePersistence.getDefault().openFiles;
+        if (openFiles.isEmpty()) {
+            sheetPane.getTabs().add(new TabSheet(SheetManager.getDefault().of(NotebookPersistence.INSTANCE.of())));
+        } else {
+            openFiles.forEach(file
+                    -> SheetManager.getDefault().open(NotebookPersistence.INSTANCE.of(file.path().toFile()))
+            );
+        }
+    }
+
+    private void shutdown() {
+        var files = FilePersistence.getDefault();
+        files.reset();
         sheetPane.getTabs().stream()
                 .filter(t -> t instanceof TabSheet)
-                .forEach(t -> ((TabSheet) t).close());
+                .map(t -> (TabSheet) t)
+                .filter(t -> null != t.sheet.getNotebook().getFile())
+                .forEach(t -> {
+                    files.add(new FilePersistence.OpenFile(t.sheet.getNotebook().getFile().toPath()));
+                    t.close(false);
+                });
+        SheetManager.getDefault().getRecentFiles().stream().forEach(rf -> {
```

Comment:
When there are no intermediate operation on the stream and the stream is only used for forEach then you can avoid the stream creation...
```suggestion
        SheetManager.getDefault().getRecentFiles().forEach(rf -> {
```